test/org/sagebionetworks/bridge/play/controllers/SubpopulationControllerTest.java

```diff
@@ -0,0 +1,190 @@
+        
+        when(study.getStudyIdentifier()).thenReturn(STUDY_IDENTIFIER);
+        doReturn(session).when(controller).getAuthenticatedSession();
+        doReturn(user).when(session).getUser();
+        doReturn(STUDY_IDENTIFIER).when(session).getStudyIdentifier();
+        when(studyService.getStudy(STUDY_IDENTIFIER)).thenReturn(study);
+    }
+
+    @Test
+    public void getAllSubpopulations() throws Exception {
+        Http.Context context = TestUtils.mockPlayContext();
+        Http.Context.current.set(context);
+        
+        Result result = controller.getAllSubpopulations();
+        
+        assertEquals(200, result.status());
+        String json = Helpers.contentAsString(result);
+        JsonNode node = BridgeObjectMapper.get().readTree(json);
+        assertNotNull(node.get("items"));
+        assertNotNull(node.get("total"));
+        assertEquals("ResourceList", node.get("type").asText());
+        
+        verify(studyService).getStudy(STUDY_IDENTIFIER);
+        verify(subpopService).getSubpopulations(study);
+    }
+    
+    @Test
+    public void createSubpopulation() throws Exception {
+        String json = "{\"name\":\"Name\",\"description\":\"Description\",\"required\":true,\"minAppVersion\":2,\"maxAppVersion\":10,\"allOfGroups\":[\"requiredGroup\"],\"noneOfGroups\":[\"prohibitedGroup\"]}";
+        Http.Context context = TestUtils.mockPlayContextWithJson(json);
+        Http.Context.current.set(context);
+        
+        Subpopulation createdSubpop = Subpopulation.create();
+        createdSubpop.setGuid("AAA");
+        createdSubpop.setVersion(1L);
+        doReturn(createdSubpop).when(subpopService).createSubpopulation(eq(study), captor.capture());
+
+        Result result = controller.createSubpopulation();
+        assertEquals(201, result.status());
+        String responseJSON = Helpers.contentAsString(result);
+        JsonNode node = BridgeObjectMapper.get().readTree(responseJSON);
+        assertEquals("AAA", node.get("guid").asText());
+        assertEquals(1L, node.get("version").asLong());
+        assertEquals("GuidVersionHolder", node.get("type").asText());
+        
+        Subpopulation created = captor.getValue();
+        assertEquals("Name", created.getName());
+        assertEquals("Description", created.getDescription());
+        assertTrue(created.isRequired());
+        assertEquals((Integer)2, created.getMinAppVersion());
+        assertEquals((Integer)10, created.getMaxAppVersion());
+        assertEquals(Sets.newHashSet("requiredGroup"), created.getAllOfGroups());
+        assertEquals(Sets.newHashSet("prohibitedGroup"), created.getNoneOfGroups());
+    }
+    
+    @Test
+    public void updateSubpopulation() throws Exception {
+        String json = "{\"name\":\"Name\",\"description\":\"Description\",\"required\":true,\"minAppVersion\":2,\"maxAppVersion\":10,\"allOfGroups\":[\"requiredGroup\"],\"noneOfGroups\":[\"prohibitedGroup\"]}";
```

Comment:
Test mismatched Guid in update? The guid is returned to the client, so they will very likely have it in their JSON when they call update()